bin/reth/src/node/mod.rs

```diff
@@ -1,48 +1,178 @@
+                "5b4590a9905fa1c9cc273f32e6dc63b4c512f0ee14edc6fa41c26b416a7b5d58"
+            )),
+        })?;
         pipeline.run(db.clone()).await?;
 
-        info!("Finishing");
+        info!("Finishing up");
         Ok(())
     }
 }
+
+/// Opens up an existing database or creates a new one at the specified path.
+fn init_db<P: AsRef<Path>>(path: P) -> eyre::Result<Env<WriteMap>> {
+    std::fs::create_dir_all(path.as_ref())?;
+    let db = reth_db::mdbx::Env::<reth_db::mdbx::WriteMap>::open(
+        path.as_ref(),
+        reth_db::mdbx::EnvKind::RW,
+    )?;
+    db.create_tables()?;
+
+    Ok(db)
+}
+
+/// Write the genesis block if it has not already been written
+fn init_genesis<DB: Database>(db: Arc<DB>) -> Result<(), reth_db::Error> {
+    let tx = db.tx_mut()?;
+    let has_block = tx.cursor::<tables::CanonicalHeaders>()?.first()?.is_some();
+    if has_block {
+        debug!("Genesis already written, skipping.");
+        return Ok(())
+    }
+
+    debug!("Writing genesis block.");
+
+    // TODO: Should be based on a chain spec
+    let mut genesis = Header::default();
+    genesis.gas_limit = 5000;
+    genesis.difficulty = U256::from(0x400000000usize);
+    genesis.nonce = 0x0000000000000042;
+    genesis.extra_data =
+        Bytes::from_str("11bbe8db4e347b4e8c937c1c8370e4b5ed33adb3db69cbdb7a38e1e50b1b82fa")
+            .unwrap()
+            .0;
+    genesis.state_root =
+        H256(hex!("d7f8974fb5ac78d9ac099b9ad5018bedc2ce0a72dad1827a1709da30580f0544"));
+    let genesis = genesis.seal();
+
+    // Insert genesis
+    tx.put::<tables::CanonicalHeaders>(genesis.number, genesis.hash())?;
+    tx.put::<tables::Headers>(genesis.num_hash().into(), genesis.as_ref().clone())?;
+    tx.put::<tables::HeaderNumbers>(genesis.hash(), genesis.number)?;
+    tx.put::<tables::CumulativeTxCount>(genesis.num_hash().into(), 0)?;
+    tx.put::<tables::HeaderTD>(genesis.num_hash().into(), genesis.difficulty.into())?;
+    tx.commit()?;
+
+    // TODO: Account balances
+    Ok(())
+}
+
+// TODO: This should be based on some external config
+fn network_config<DB: Database>(db: Arc<DB>) -> NetworkConfig<ProviderImpl<DB>> {
+    NetworkConfig::builder(Arc::new(ProviderImpl::new(db)), rng_secret_key())
+        .boot_nodes(mainnet_nodes())
+        .build()
+}
+
+// TODO: This should be based on some external config
+fn consensus_config() -> reth_consensus::Config {
+    reth_consensus::Config::default()
+}
+
+/// Starts the networking stack given a [NetworkConfig] and returns a handle to the network.
+async fn start_network<C>(config: NetworkConfig<C>) -> Result<NetworkHandle, NetworkError>
+where
+    C: BlockProvider + HeaderProvider + 'static,
+{
+    let client = config.client.clone();
+    let (handle, network, _txpool, eth) =
+        NetworkManager::builder(config).await?.request_handler(client).split_with_handle();
+
+    tokio::task::spawn(network);
+    //tokio::task::spawn(txpool);
```

Comment:
@mattsse  @onbjerg Feels like we should have an `async into_handle()` function on the Config?